onnxscript/rewriter/ort_fusions/attention.py

```diff
@@ -0,0 +1,287 @@
+                qkv_weight,
+            )
+        if no_match(qkv_bias, ["Dh"]):
+            return check_result.fail(
+                f"Shape mismatch: {qkv_bias} does not match expected dimensions ['Dh']",
+                qkv_bias,
+            )
+        if no_match(query_mm_sliced, ["B", "S", "Dh_q"]):
+            return check_result.fail(
+                f"Shape mismatch: {query_mm_sliced} does not match expected dimensions ['B', 'S', 'Dh_q']",
+                query_mm_sliced,
+            )
+        if no_match(key_mm_sliced, ["B", "S", "Dh_k"]):
+            return check_result.fail(
+                f"Shape mismatch: {key_mm_sliced} does not match expected dimensions ['B', 'S', 'Dh_k']",
+                key_mm_sliced,
+            )
+        if no_match(value_mm_sliced, ["B", "S", "Dh_v"]):
+            return check_result.fail(
+                f"Shape mismatch: {value_mm_sliced} does not match expected dimensions ['B', 'S', 'Dh_v']",
+                value_mm_sliced,
+            )
+
+        # Ensure Dh = Dh_q + Dh_k + Dh_v
+        Dh = self.bindings.get("Dh")
+        Dh_q = self.bindings.get("Dh_q")
+        Dh_k = self.bindings.get("Dh_k")
+        Dh_v = self.bindings.get("Dh_v")
+
+        if (
+            not isinstance(Dh, int)
+            or not isinstance(Dh_q, int)
+            or not isinstance(Dh_k, int)
+            or not isinstance(Dh_v, int)
+        ):
+            return check_result.fail(
+                "Could not determine the hidden sizes of query, key, and value.",
+            )
+
+        if Dh != Dh_q + Dh_k + Dh_v:  # type: ignore[operator]
+            return check_result.fail(
+                f"Hidden size of query, key and value do not add up to hidden size: {Dh} != {Dh_q} + {Dh_k} + {Dh_v}",
+            )
+
+        # TODO: Add mask check once mask is added to the pattern
+        return check_result
+
+    def rewrite(
+        self,
+        op,
+        input,
+        qkv_weight,
+        qkv_bias,
+        # mask_index,
+        past,
+        # attention_bias,
+        num_heads,
+        # scale,
+        **_,
+    ):
+        # Use bindings to get the values of Dh_q, Dh_k, and Dh_v
+        # and construct qkv_hidden_sizes
+        Dh_q = self.bindings.get("Dh_q")
+        Dh_k = self.bindings.get("Dh_k")
+        Dh_v = self.bindings.get("Dh_v")
+        qkv_hidden_sizes = [Dh_q, Dh_k, Dh_v]
+
+        if self._has_past:
+            attention, present = op.Attention(
+                input,
+                qkv_weight,
+                qkv_bias,
+                None,  # mask_index
+                past,
+                # attention_bias,
+                # past_sequence_length
+                num_heads=num_heads,
+                qkv_hidden_sizes=qkv_hidden_sizes,
+                # scale=scale,
+                _domain="com.microsoft",
+                _outputs=2,
+            )
+            # Return present output first as it captures the complete rewrite pattern graph
```

Comment:
This comment should be in the return statement of the pattern method above. Here, I suggest something like "Use same output ordering as in pattern"